### PR TITLE
frint-props: API docs

### LINKED ISSUE #15: frint-props: create README with API docs

### PR DESCRIPTION
**Pull request by fahad19**
Closes #15

### CODE REVIEW

File: packages/frint-props/README.md [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+- [Guide](#guide)
+  - [Installation](#installation)
+  - [Concepts](#concepts)
+  - [Usage](#usage)
+  - [Write your own functions](#write-your-own-functions)
+- [API](#api)
+  - [withDefaults](#withdefaults)
+  - [withState](#withstate)
+  - [withState](#withstate-1)
```
Comment (discosultan) [inline packages/frint-props/README.md]: Why is `withState` linked twice? It seems like a typo somewhere between `withDefaults` and `withState` descriptions.

Comment (fahad19) [reply packages/frint-props/README.md]: investigating... this block is generated automatically by SublimeText's MarkdownTOC plugin. not sure why it thinks there are two `withState`s.

Comment (fahad19) [reply packages/frint-props/README.md]: well.. I did add two `withState`s myself. removing it...

File: packages/frint-props/README.md [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+> withDefaults(defaultProps)
+
+### Arguments
+
+1. `defaultProps`: Default props to start the stream with
+
+<!-- -->
+
+## withState
```
Comment (nunorfpt) [inline packages/frint-props/README.md]: Is this part duplicated from the withState defined at line 271? Also the example of this part belongs to the withDefaults section above this one

Comment (fahad19) [reply packages/frint-props/README.md]: yep! now removed.

File: packages/frint-props/README.md [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+<script src="https://unpkg.com/frint-props@latest/dist/frint-props.min.js"></script>
+
+<script>
+  // available as `window.FrintProps`
+</script>
+```
+
+## Concepts
+
+The package consists of multiple functions, that enables you to compose your props as an RxJS Observable.
```
Comment (discosultan) [inline packages/frint-props/README.md]: I'd remove all the characters in **bold**.

> The package consists of multiple functions**,** that enable**s** you to compose your props as a**n** RxJS Observable.

File: packages/frint-props/README.md [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+There are two kinds of functions:
+
+1. That adds new props to the stream
+1. That processes the props stream further (like RxJS operators)
+
+And then there is `compose` function that accepts both kinds of functions as arguments, and returns a single function that you can use anywhere.
+
+## Usage
+
+We can start small, and have prepare a stream that only emits the props once, and it doesn't change over time:
```
Comment (discosultan) [inline packages/frint-props/README.md]: Sounds weird.

> ... and have prepare ...

File: packages/frint-props/README.md [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+
+Now that we have the Observable available as `props$`, we can subscribe to it as needed:
+
+```js
+props$.subscribe(props => console.log(props));
+```
+
+### Reactivity
+
+But we have more real world use cases, that requires our props to change over time too. We can consider using `withState` for this example:
```
Comment (discosultan) [inline packages/frint-props/README.md]: > But we have more real world use cases**,** that require**s** our props to change over time too.

File: packages/frint-props/README.md [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+```
+
+Now the `props$` Observable will emit an object with these keys:
+
+* `counter` (`Integer`): The counter value
+* `setCounter(n)` (`Function`): Call this function to update `counter` value
+
+### Composition
+
+You can compose multiple functions together, to generated a combined stream of all props. For that, we can use the `compose` function:
```
Comment (discosultan) [inline packages/frint-props/README.md]: > You can compose multiple functions together**,** to generate**d** a combined stream of all props.

File: packages/frint-props/README.md [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+
+```js
+const app = new App();
+
+const props$ = compose(
+  withFoo()
+)(app);
+```
+
+The argument `app` can now accessed inside our function like this:
```
Comment (discosultan) [inline packages/frint-props/README.md]: The word "be" should be added.

File: packages/frint-props/README.md [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+}
+```
+
+The way some of our functions are designed in this repository, the returned functions expect to receive the same arguments as our `observe` higher-order component receives in `frint-react`, which are: `app` and `parentProps$`.
+
+### Operator functions
+
+Besides just adding new props, functions can also take care of processing the stream further just like how RxJS operators work.
+
+We can create a function, that will check if there is any `foo` prop, and then capitalize it:
```
Comment (discosultan) [inline packages/frint-props/README.md]: Also the comma here doesn't sound right.

**Pull request merged**